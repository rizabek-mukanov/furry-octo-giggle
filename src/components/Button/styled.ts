import styled from 'styled-components';

export const StyledButton = styled.button`
  background-color: ${({ theme }) => theme.colors.primary};
  border-radius: ${({ theme }) => theme.borderRadius};
  border: none;
  color: ${({ theme }) => theme.colors.white};
  height: 42px;
  width: 100%;
`;
