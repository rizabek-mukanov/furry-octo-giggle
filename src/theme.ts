import { Theme } from 'styled-components';
export const theme: Theme = {
  borderRadius: '10px',
  colors: {
    primary: '#2F80ED',
    secondary: '#102C51',
    gray: '#BDBDBD',
    darkGray: '#828282',
    white: '#f2f2f2',
    borderColor: '#CFCFCF',
    text: '#102C51',
    background: '#f2f2f2',
  },
  fontSize: {
    h1: '30px',
    h2: '24px',
    h3: '18px',
    h4: '14px',
    h5: '10px',
  },
};
