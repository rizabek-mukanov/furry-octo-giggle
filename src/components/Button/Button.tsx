import React from 'react';
import { StyledButton } from './styled';
import { ButtonProps } from 'index.d';

// Not sure what this button should do
// In google document it says:
// When the users change the value of any of the inputs, the monthly deposit value is calculated and displayed to them.

function Button({ children }: ButtonProps) {
  return <StyledButton data-testid="finish-button"> {children}</StyledButton>;
}

export default Button;
