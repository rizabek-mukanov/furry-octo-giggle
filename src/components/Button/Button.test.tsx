import React from 'react';
import Button from './Button';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from 'theme';

it('renders correct child', () => {
  const { getByTestId } = render(
    <ThemeProvider theme={theme}>
      <Button>Finish</Button>
    </ThemeProvider>
  );
  expect(getByTestId('finish-button')).toHaveTextContent('Finish');
});

it('matches snapshot', () => {
  const { asFragment } = render(
    <ThemeProvider theme={theme}>
      <Button>Finish</Button>
    </ThemeProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});
