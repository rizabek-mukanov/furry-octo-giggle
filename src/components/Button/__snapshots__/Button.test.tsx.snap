// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background-color: #2F80ED;
  border-radius: 10px;
  border: none;
  color: #f2f2f2;
  height: 42px;
  width: 100%;
}

<button
    class="c0"
    data-testid="finish-button"
  >
     Finish
  </button>
</DocumentFragment>
`;
